osu.Game/Graphics/UserInterface/OsuTextBox.cs

```diff
@@ -112,7 +125,72 @@ protected override void OnCaretMoved(bool selecting)
         {
             base.OnCaretMoved(selecting);
 
-            caretMovedSample?.Play();
+            if (!selecting)
+                caretMovedSample?.Play();
+        }
+
+        protected override void OnTextSelectionChanged(TextSelectionType selectionType)
+        {
+            base.OnTextSelectionChanged(selectionType);
+
+            if (selectionType == TextSelectionType.Word)
+            {
+                if (!selectionStarted)
+                    playSelectSample(selectionType);
+                else
+                    playSelectSample();
+            }
+            else
+            {
+                playSelectSample(selectionType);
+            }
+
+            selectionStarted = true;
+        }
+
+        protected override void OnTextDeselected()
+        {
+            if (selectionStarted)
+                playSelectSample(TextSelectionType.Deselect);
+
+            selectionStarted = false;
+
+            base.OnTextDeselected();
+        }
+
+        private void playSelectSample(TextSelectionType selectionType = TextSelectionType.Character)
+        {
+            if (Time.Current < sampleLastPlaybackTime + 15) return;
+
+            SampleChannel? channel;
+            double pitch = 0.98 + RNG.NextDouble(0.04);
+
+            switch (selectionType)
+            {
+                case TextSelectionType.All:
+                    channel = selectAllSample?.GetChannel();
+                    break;
+
+                case TextSelectionType.Word:
+                    channel = selectWordSample?.GetChannel();
+                    break;
+
+                case TextSelectionType.Deselect:
+                    channel = deselectSample?.GetChannel();
+                    break;
+
+                default:
+                    channel = selectCharSample?.GetChannel();
+                    pitch += (SelectedText.Length / (float)Text.Length) * 0.15f;
```

Comment:
Possible division by 0 here, resulting in `NaN` pitch. Shockingly, this doesn't crash, but instead plays the sound high-pitched. It's probably not intended for selection to change and `Text.Length` to be `0` (framework-side bug) but I think this should be guarded against.

Also, pitch is `double`, but this is casting to `float`.

Reply:
How were you able to trigger a selection change with `Text.Length` being zero? It's probably better to ensure `OnTextSelectionChanged` is never called when `Text.Length` is zero than guarding here?